src/com/reco1l/osu/ui/Preference.kt

```diff
@@ -0,0 +1,308 @@
+package com.reco1l.osu.ui
+
+import android.annotation.SuppressLint
+import android.content.Context
+import android.content.res.TypedArray
+import android.os.Parcel
+import android.os.Parcelable
+import android.text.TextUtils
+import android.util.AttributeSet
+import android.view.KeyEvent
+import android.view.inputmethod.EditorInfo.IME_ACTION_DONE
+import android.widget.EditText
+import androidx.core.content.res.TypedArrayUtils
+import androidx.preference.Preference
+import androidx.preference.PreferenceViewHolder
+import androidx.preference.R
+import com.reco1l.toolkt.android.hideKeyboard
+
+
+/// Select
+
+@SuppressLint("RestrictedApi")
+open class SelectPreference(context: Context, attrs: AttributeSet?, defStyleAttr: Int, defStyleRes: Int): Preference(context, attrs, defStyleAttr, defStyleRes) {
+
+    constructor(context: Context) : this(context, null)
+    constructor(context: Context, attrs: AttributeSet?, defStyleAttr: Int) : this(context, attrs, defStyleAttr, 0)
+    constructor(context: Context, attrs: AttributeSet?) : this(context, attrs, TypedArrayUtils.getAttr(context, R.attr.dialogPreferenceStyle, android.R.attr.dialogPreferenceStyle))
```

Comment:
This is using a restricted API, as suppressed by the `SuppressLint` annotation, but is this really needed? The only usage of this constructor is by line 25 (`constructor(context: Context) : this(context, null)`). I believe this can be removed if it's not necessary.